test/configuration/index.test.ts

```diff
@@ -174,24 +174,26 @@ describe('test configuration', () => {
 		});
 
 		it('requesting nested values', () => {
-			process.env['Nested.foo'] = "another bar";
+			const { dotNotationSeparator } = defaultOptions;
+			process.env['Nested' + dotNotationSeparator + 'foo'] = "another bar";
```

Comment:
const envName = 'Nested' + dotNotationSeparator + 'foo';
and use envName for selects?
in 'set Nested values'  too